builder/dockerfile/builder.go

```diff
@@ -196,6 +199,11 @@ func (b *Builder) dispatchDockerfileWithCancellation(dockerfile *parser.Result)
 		// TODO: get this from dispatch
 		imageID = b.image
 
+		if b.Aux != nil {
+			if err := b.Aux.Emit(types.BuildResult{ID: imageID}); err != nil {
+				return "", err
+			}
+		}
```

Comment:
> This is going to Emit() an ID for every instruction in the Dockerfile, is that what you want?

Yes, based on @tonistiigi's suggestion in https://github.com/moby/moby/pull/32406#issuecomment-292236996 followed up in https://github.com/moby/moby/pull/32406#issuecomment-292296637

> I don't know if this should be part of dispatch, it might be more appropriate in api/server/backend/build/backend.go.

I think that's only possible if we don't do as @tonistiigi suggested.